### PR TITLE
Added Ethereum Mist

### PR DESCRIPTION
**Pull request by rushsteve1**
The [Mist client](https://github.com/ethereum/mist) as a flatpak for easy installation on all platforms!
This is something that I personally wanted, so I went ahead and did it.
This is my first flatpak, so let me know if there are any issues, I tried to use the existing examples on Flathub as reference (specifically the ARK wallet).

### CODE REVIEW

File: org.ethereum.Mist.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <developer_name>Ethereum</developer_name>
+    <summary>Ethereum Dapp Browser</summary>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>GPL-3.0-only</project_license>
+    <url type="homepage">https://ethereum.org</url>
+
+    <description>
+        <p>The Mist browser is the tool of choice to browse and use Ðapps.</p>
+    </description>
+
```
Comment (TingPing) [inline org.ethereum.Mist.appdata.xml]: Screenshots are appreciated.

File: org.ethereum.Mist.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    "base-version": "stable",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "mist",
+    "appstream-compose": true,
+    "separate-locales": false,
+    "finish-args": [
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (TingPing) [inline org.ethereum.Mist.json]: Electron doesn't support wayland.

File: org.ethereum.Mist.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        "--share=network",
+        "--filesystem=xdg-download"
+    ],
+    "modules": [
+        {
+            "name": "mist",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -D org.ethereum.Mist.desktop /app/share/applications/org.ethereum.Mist.desktop",
+                "install -D org.ethereum.Mist.appdata.xml /app/share/appdata/org.ethereum.Mist.appdata.xml",
```
Comment (TingPing) [inline org.ethereum.Mist.json]: These should not be executable, `install -Dm644 ...` to set the mode.

File: org.ethereum.Mist.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                {
+                    "type": "script",
+                    "dest-filename": "mist",
+                    "commands": [
+                        "exec env TMPDIR=$XDG_CACHE_HOME /app/share/Mist/mist $@"
+                    ]
+                },
+                {
+                    "type": "file",
+                    "url": "https://github.com/ethereum/mist/releases/download/v0.9.3/Mist-linux64-0-9-3.zip",
```
Comment (TingPing) [inline org.ethereum.Mist.json]: Create a `flathub.json` file containing:

```json
{
  "only-arches": ["x86_64"]
}

File: org.ethereum.Mist.json (hunk 2 above)
Comment (rushsteve1) [reply org.ethereum.Mist.json]: Huh. I didn't know that. Are they working on it?

File: org.ethereum.Mist.json (hunk 4 above)
Comment (rushsteve1) [reply org.ethereum.Mist.json]: Is there a way to have the manifest download the 32 bit version when building for that platform?

File: org.ethereum.Mist.json (hunk 2 above)
Comment (TingPing) [reply org.ethereum.Mist.json]: There is work in Chromium to get Wayland support which probably will trickle into Electron. Give it a year.

File: org.ethereum.Mist.json (hunk 4 above)
Comment (TingPing) [reply org.ethereum.Mist.json]: Yes there is an `only-arches` property for sources. See `man flatpak-manifest`

Comment (TingPing) [reply org.ethereum.Mist.json]: You will still want that file because they probably don't ship arm binaries. Just add `"i386"` to the list.

**Comment by rushsteve1** [comment]
I fixed the issues with that last commit. Is there anything else wrong?

**Comment by TingPing** [comment]
bot, build org.ethereum.Mist

**Comment by TingPing** [comment]
> Failed to download sources: module mist: Wrong sha256 checksum for Mist-linux32-0-9-3.zip, expected 72425a83378539c9b774c761b8dda10af04f6cbb6f5f37f6a4fa944e4159dfc7, was a3fd569d6916e819b3c21b65daddef2f3840d6edba2602c0125f6701f5f8833e

**Comment by rushsteve1** [comment]
Sorry about that. Copied the wrong SHA off the releases page.

**Comment by nedrichards** [comment]
bot, build org.ethereum.Mist

**Comment by rushsteve1** [comment]
Should be fixed now. The 32 and 64 bit versions unpack to different folder names.
Also who has permissions to cause the bot to build? It would be convenient if I didn't have to wait for someone to test it.

**Comment by nedrichards** [comment]
bot, build org.ethereum.Mist

**Comment by nedrichards** [comment]
Whilst you're in review - only reviewers and admins can do that - which is annoying but part of the per repo permissions. Once the app is accepted you will be able to do that as well for future PRs to update or add features.

**Comment by rushsteve1** [comment]
Is there something wrong with BuildBot? It didn't build after that last command, and some of the builds in the queue have been going for really long times. As in multiple hours.

**Comment by TingPing** [comment]
@rushsteve1 At a glance the worker is hung up importing the GNOME runtime which is large.

**Comment by nedrichards** [comment]
Yeah, it takes a while to import large runtimes into the repo - mostly in the generation of static deltas - which are an important part of making downloads small. This is a high CPU workload and has recently got slower for reasons you might be aware of...

**Comment by rushsteve1** [comment]
Well the build finally finished and passed. Anything else need fixing or is this ready?

**Comment by TingPing** [comment]
Repository has been created: https://github.com/flathub/org.ethereum.Mist

**Comment by TingPing** [comment]
@rushsteve1 BTW you were sent an invite to the repo when it was created and just have yet to accept it.

**Comment by rushsteve1** [comment]
Oh. Thanks @TingPing sorry, I don't check that email often.

**Pull request closed without merging**